language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/CheckExpressionNodeContext.java

```diff
@@ -58,12 +63,24 @@ public CheckExpressionNodeContext() {
                 || node.parent().kind() == SyntaxKind.OBJECT_FIELD) {
             completionItems.addAll(CompletionUtil.route(ctx, node.parent()));
         } else {
-            /*
-            We add the action keywords in order to support the check action context completions
-             */
-            completionItems.addAll(this.actionKWCompletions(ctx));
-            completionItems.addAll(this.expressionCompletions(ctx));
-            completionItems.add(new SnippetCompletionItem(ctx, Snippet.STMT_COMMIT.get()));
+
+            NonTerminalNode nodeAtCursor = ctx.getNodeAtCursor();
+            if (QNameReferenceUtil.onQualifiedNameIdentifier(ctx, nodeAtCursor)) {
+                Predicate<Symbol> filter = symbol -> symbol.kind() == SymbolKind.VARIABLE
+                        || symbol.kind() == SymbolKind.FUNCTION
+                        || symbol.kind() == SymbolKind.TYPE_DEFINITION
+                        || symbol.kind() == SymbolKind.CLASS;
+                List<Symbol> types = QNameReferenceUtil.getModuleContent(ctx,
+                        (QualifiedNameReferenceNode) nodeAtCursor, filter);
+                completionItems.addAll(getCompletionItemList(types, ctx));
```

Comment:
Shall we use the `QNameReferenceUtil.QNameReferenceUtil` instead?

Reply:
Done